element/index.js

```diff
@@ -65,3 +66,20 @@ export function renderToString( element ) {
 
 	return renderToStaticMarkup( element );
 }
+
+export function concatValues( value1, value2 ) {
+	const toArray = value => Array.isArray( value ) ? Children.toArray( value ) : [ value ];
+
+	return toArray( value1 )
+		.concat( toArray( value2 ) )
+		.map( ( elt, index ) => {
+			if ( isString( elt ) ) {
+				return elt;
+			}
+
+			return {
+				...elt,
+				key: index
+			};
+		} );
+}
```

Comment:
Evidenced by my confusion on the behavior of this function, would be good to add some JSDoc and test cases.